go/merkletree2/position.go

```diff
@@ -103,25 +127,47 @@ func (t *Config) getDeepestPositionForKey(k Key) (*Position, error) {
 	return &p, nil
 }
 
-// getSiblingPositionsOnPathToKey returns a slice of positions, in descending
-// order by level (siblings farther from the root come first) and in
-// lexicographic order within each level.
-func (t *Config) getSiblingPositionsOnPathToKey(k Key) ([]Position, error) {
-	p, err := t.getDeepestPositionForKey(k)
-	if err != nil {
-		return nil, err
+// getDeepestPositionAtLevelAndSiblingsOnPathToKey returns a slice of positions,
+// in descending order by level (siblings farther from the root come first) and
+// in lexicographic order within each level. The first position in the slice is
+// the position at level lastLevel on a path from the root to k (or the deepest
+// possible position for such key if latLevel is greater than that). The
+// following positions are all the siblings of the nodes on the longest possible
+// path from the root to the key k with are at levels from lastLevel (excluded)
+// to firstLevel (included).
+// See TestGetDeepestPositionAtLevelAndSiblingsOnPathToKey for sample outputs.
+func (t *Config) getDeepestPositionAtLevelAndSiblingsOnPathToKey(k Key, lastLevel int, firstLevel int) (sibs []Position) {
+
+	maxLevel := t.keysByteLength * 8 / int(t.bitsPerIndex)
+	if lastLevel > maxLevel {
+		lastLevel = maxLevel
+	}
+
+	// first, shrink the key for efficiency
+	var bytesNecessary int
+	if lastLevel*int(t.bitsPerIndex)%8 == 0 {
```

Comment:
```go
bytesNecessary = lastLevel * int(t.bitsPerIndex) 
if lastLevel*int(t.bitsPerIndex)%8 != 0 {
    bytesNecessary++
}
```

Reply:
Is this for readability? or are there other reasons for the change? Also the first line should be 
` bytesNecessary = lastLevel * int(t.bitsPerIndex) / 8`  right?

Reply:
readability